### ISSUE TITLE
Missing consumer rules on 6.4.4

### ISSUE THREAD

**Issue by matteinn** [GetStream/stream-chat-android#5359]
**Describe the bug**

I recently bumped the SDK version from 6.3.2 to the latest 6.4.4 and started getting R8 errors.

> ERROR: Missing classes detected while running R8. Please add the missing classes or apply additional keep rules that are generated in /Users/vagrant/git/androidApp/build/outputs/mapping/prodRelease/missing_rules.txt.
> ERROR: R8: Missing class io.getstream.chat.android.previewdata.PreviewMessageData (referenced from: void io.getstream.chat.android.compose.ui.components.reactionpicker.ComposableSingletons$ReactionsPickerKt$lambda-2$1.invoke(androidx.compose.runtime.Composer, int))
> Missing class io.getstream.chat.android.previewdata.PreviewUserData (referenced from: io.getstream.chat.android.compose.state.userreactions.UserReactionItemState io.getstream.chat.android.compose.previewdata.PreviewUserReactionData.user1Reaction(androidx.compose.runtime.Composer, int) and 3 other contexts)

This is the suggested workaround but it's not working, furthermore I can't see that code when browsing the dependencies used by my app:
```
# Please add these rules to your existing keep rules in order to suppress warnings.
# This is generated automatically by the Android Gradle plugin.
-dontwarn io.getstream.chat.android.previewdata.PreviewMessageData
-dontwarn io.getstream.chat.android.previewdata.PreviewUserData
```

Can you please advise on what to do?

You have [the same issue reported by your CI](https://github.com/GetStream/stream-chat-android/actions/runs/10247791519/job/28347705815), please don't ignore it.

**SDK version**
```
    implementation("io.getstream:stream-chat-android-state:6.4.4")
    implementation("io.getstream:stream-chat-android-offline:6.4.4")
    implementation("io.getstream:stream-chat-android-compose:6.4.4")
    implementation("io.getstream:stream-android-push-firebase:1.1.8")
```

**To Reproduce**
1. Run the app with R8 enabled

**Expected behavior**
App is successfully built

**Device:**
 - Vendor and model: doesn't matter
 - Android version: doesn't matter

**Screenshots**
None

**Comment by matteinn** [comment]
Thank you!
Unfortunately that's not working, and at this point I'm not sure it will work with the new release too 🧐 
The weird thing is that the `previewdata` module is not showing up in the External Libraries section of Android Studio, that might be the reason why that rule added to my app isn't changing anything.

**Comment by JcMinarro** [comment]
I think it can be explained by [this](https://github.com/GetStream/stream-chat-android/pull/5358/files#diff-e2ad31bb5f10a51e9d2ef59a777bc4e5944023da3292c579d5eaef98751df5f7R12)
@skydoves could you check if we need anything else to fix R8/Proguard issues related to the preview data artifact??

**Comment by matteinn** [comment]
This is fixed in 6.5.0
